src/nameutil.rs

```diff
@@ -125,6 +125,30 @@ pub fn lib_name_to_toml(name: &str) -> String {
     name.to_string().replace("-", "_").replace(".", "_")
 }
 
+pub fn shared_lib_name_to_link_name(name: &str) -> &str {
+    let mut s = name;
+
+    if s.starts_with("lib") {
+        s = &s[3..];
+    }
+
+    let offset = s.find(".so").unwrap_or_else(|| s.len());
+    s = &s[..offset];
+
+    s
+}
+
+pub fn shared_libs_to_links(shared_libs: &[String]) -> String {
+    // https://github.com/rust-lang/cargo/issues/4533
+    // links = ["foo", "bar"] or ['foo', 'bar'] is not supported.
+    // Pick the first element
+    if !shared_libs.is_empty() {
+        return format!("\"{}\"", shared_lib_name_to_link_name(&shared_libs[0]));
+    }
```

Comment:
Sorry, should have suggested:

```suggestion
if let Some(shared_lib) = shared_libs.iter().next() {
    return format!("\"{}\"", shared_lib_name_to_link_name(&shared_lib));
}
```

Reply:
Or even better: if let Some(item) = vec.get(0)

No need for an iterator.

Reply:
True!

Reply:
thanks for the suggestions!